src/controllers/collection.controller.ts

```diff
@@ -0,0 +1,151 @@
+import { Locals, NextFunction, Request, Response } from 'express';
+import { NO_CONTENT, NOT_FOUND } from 'http-status';
+import { CreationAttributes, Op } from 'sequelize';
+import { URLSearchParams } from 'url';
+import { CreatePagination, SafeController, Pagination } from '@/controllers/decorators';
+import Collection from '@/models/collection';
+import Product from '@/models/product';
+import env from '@/configs/env';
+
+// eslint-disable-next-line max-len
+type LoadedCollectionResponse<T = any> = Response<
+  T,
+  { collection: Collection; [index: string]: unknown }
+>;
+
+class CollectionController {
+  @SafeController
+  static async load(req: Request, res: Response, next: NextFunction) {
+    const { collectionId } = req.params;
+    const collection = await Collection.findByPk(collectionId);
+
+    if (!collection) throw res.status(NOT_FOUND).send('Collections not found');
+
+    res.locals = { ...res.locals, collection };
+    next();
+  }
+
+  @SafeController
+  @CreatePagination
+  static async show(
+    req: Request,
+    res: Response<
+      any,
+      Record<string, any> &
```

Comment:
try to avoid to use `Record<string, any> ` Location should cover this

Reply:
I'm not sure I understand exactly how to describe it..